src/MICore/Debugger.cs

```diff
@@ -517,6 +517,11 @@ protected bool IsCoreDump
 
         public async Task<Results> CmdTerminate()
         {
+            if (ProcessState == ProcessState.Running)
+            {
+                await CmdBreak();
+            }
```

Comment:
This isn't a bug in your stuff, but the cmdbreak that uses signals returns before the process breaks. We really need to get that fixed at some point.